### PR TITLE
fix: correct slots_per_trial limits for hp search [DET-7956]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Fix the slots_per_trial default value bug.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx [hunk 1]
```diff
@@ -446,13 +447,12 @@ const useModalHyperparameterSearch = ({
             </SelectFilter>
           </Form.Item>
           <Form.Item
-            initialValue={experiment.configRaw?.resources?.slots_per_trial ??
-              maxSlots === 0 ? 0 : 1}
```
Comment (hkang1) [inline webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx:450]: this ternary should have parentheses around it, as it's evaluating to `0` whenever `maxSlots` is 0, even when `slots_per_trial` is defined as greater than 0.

Comment (hkang1) [reply webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx:450]: talking to backend, this value should never be 0, which causes scheduling problems.

File: webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx [hunk 2]
```diff
@@ -357,6 +357,7 @@ const useModalHyperparameterSearch = ({
   }, [ hyperparameters, modalError, searcher ]);
 
   const searcherPage = useMemo((): React.ReactNode => {
+    console.log('spt', experiment.configRaw?.resources?.slots_per_trial);
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx:360]: nit: remove console.log

**Review by EmilyBonar — APPROVED** [review]
LGTM! Thanks for figuring this out and fixing it!

File: webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx (hunk 2 above)
Comment (hkang1) [reply webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx:360]: oops

**Pull request merged**